### PR TITLE
Implemented ability to toggle visibility of actors via SHOW_ACTOR action

### LINKED ISSUE #127: SHOW_ACTOR
## Description
Implement the SHOW_ACTOR action. It should take a bool to indicate whether the actor is visible (true) or not (false)

## Acceptance Criteria

This story will be finished when the SHOW_ACTOR action has been demonstrated to hide and show the current actor.

### PR DESCRIPTION
**Pull request by ViMaSter**
## Summary
Adds `&SHOW_ACTOR:true` and `&SHOW_ACTOR:false` actions to toggle visibility of the currently displayed actor.

The placeholder for the logic to toggle actor visibility was in `SceneController`, but as it's tied to manipulating the sprites and things like playing emotes or talking are already inside `ActorController`, I've moved the methods there. Open for feedback on that however, if you feel it makes sense in another place!

## Testing instructions
1. Open the `Inky- TestScene`
2. Change the `StoryController` to run `/Assets/Resources/InkDialogueScripts/TestScripts/VisibilityTest`
3. Hit play and play through the script and notice Actors appearing and disappearing

## Additional information

- `[ ]` Changes UI
- `[ ]` Introduces new feature
- `[ ]` Removes existing feature
- `[x]` Has associated resource: 
  - Closes #127

### CODE REVIEW

File: unity-ggjj/Assets/Scripts/Actor/ActorController.cs [hunk 1]
```diff
@@ -302,4 +302,9 @@ public void AssignActorToSlot(string actor, int oneBasedSlotIndex)
             Debug.Log($"{exception.GetType().Name}: Actor {actor} was not found in actor dictionary");
         }
     }
+
+    public void SetVisibility(string actorName, bool shouldShow)
```
Comment (PhoebeMitchell) [inline unity-ggjj/Assets/Scripts/Actor/ActorController.cs:299]: Nit: Add summary

File: unity-ggjj/Assets/Resources/InkDialogueScripts/TestScripts/VisibilityTest.ink [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+&JUMP_TO_POSITION:2
+&SPEAK:Ross
+But I'm back now!
+
+&SHOW_ACTOR:Arin,true
+&JUMP_TO_POSITION:1
+&SPEAK:Arin
+So am I!
+
+-> END
```
Comment (PhoebeMitchell) [inline unity-ggjj/Assets/Resources/InkDialogueScripts/TestScripts/VisibilityTest.ink:43]: Write an automated test for this? 😉

File: unity-ggjj/Assets/Scripts/TextDecoder/ActionDecoder.cs [hunk 3]
```diff
@@ -508,21 +508,15 @@ private void ACTOR(ActorAssetName actorName)
         OnActionDone?.Invoke();
     }
 
-    /// <summary>Shows or hides the actor on the screen. Has to be re-done after switching a scene.</summary>
-    /// <param name="shouldShow">whether to show (`true`) or not show (`false`) an actor</param>
-    /// <example>&amp;SHOW_ACTOR:true</example>
-    /// <example>&amp;SHOW_ACTOR:false</example>
+    /// <summary>Shows or hides all actors in the scene. Has to be re-done after switching a scene.</summary>
+    /// <param name="actorName" validFiles="Assets/ScriptableObjects/Actors/*.asset">Name of the actor</param>
+    /// <param name="shouldShow">whether to show (`true`) or not show (`false`) all actors</param>
```
Comment (PhoebeMitchell) [inline unity-ggjj/Assets/Scripts/TextDecoder/ActionDecoder.cs]: UltraMegaNit: Capital W for whether

**Review by PhoebeMitchell — DISMISSED** [review]

File: unity-ggjj/Assets/Scripts/TextDecoder/ActionDecoder.cs (hunk 3 above)
Comment (ViMaSter) [reply unity-ggjj/Assets/Scripts/TextDecoder/ActionDecoder.cs]: [You mean... super mega ultra nit?](https://www.youtube.com/watch?v=31sTVI3V7go&t=505s) <sup>(Oh and: Fixed. 👀)</sup>

File: unity-ggjj/Assets/Resources/InkDialogueScripts/TestScripts/VisibilityTest.ink (hunk 2 above)
Comment (ViMaSter) [reply unity-ggjj/Assets/Resources/InkDialogueScripts/TestScripts/VisibilityTest.ink:43]: Actually a super good idea for this case, thanks for the reminder! Added one in https://github.com/Studio-Lovelies/GG-JointJustice-Unity/pull/158/commits/588588485cc1d314752f10ee069a8c9490c88c79#diff-2882ae544b7ed7df1d862321d0f4d6148ed3c5b304caf8f298b7b8510e2d24d9

**Review by notexplosive — DISMISSED** [review]

File: unity-ggjj/Assets/Scripts/Actor/ActorController.cs [hunk 4]
```diff
@@ -108,53 +108,48 @@ private ActorData FindActorDataInInventory(string actorName)
...
+    /// <param name="actorName">Name of the ActorData object to find the actor by</param>
+    /// <returns>Instance of <see cref="Actor"/> the actor is currently assigned to</returns>
     private Actor FindActorInInventory(string actorName)
     {
         var actorData = FindActorDataInInventory(actorName);
-        if (actorData != null && _actorDataToActor.ContainsKey(actorData))
+        if (actorData == null || !_actorDataToActor.ContainsKey(actorData))
         {
-            return _actorDataToActor[actorData];
+            throw new KeyNotFoundException($"No actor with the name '{actorName}' could be found; make sure both ");
```
Comment (PhoebeMitchell) [inline unity-ggjj/Assets/Scripts/Actor/ActorController.cs]: Did this sentence get cut off? "make sure both..."

File: unity-ggjj/Assets/Scripts/Actor/ActorController.cs [hunk 5]
```diff
@@ -108,53 +108,48 @@ private ActorData FindActorDataInInventory(string actorName)
...
-        return null;
+        return _actorDataToActor[actorData];
     }
 
     /// <summary>
     /// Sets the pose of the active actor.
     /// In working this is mostly the same as PlayEmotion without calling OnAnimationStarted so the system can continue without waiting for the animation to end.
     /// </summary>
-    /// <param name="pose"></param>
+    /// <param name="pose">Name of the post to execute</param>
```
Comment (PhoebeMitchell) [inline unity-ggjj/Assets/Scripts/Actor/ActorController.cs]: Spelling: post -> pose

File: unity-ggjj/Assets/Scripts/Actor/ActorController.cs [hunk 6]
```diff
@@ -108,53 +108,48 @@ private ActorData FindActorDataInInventory(string actorName)
...
+    /// <param name="actorName">Optionally name of another actor to run this animation on (defaults to <see cref="_activeActor"/> if not set)</param>
```
Comment (PhoebeMitchell) [inline unity-ggjj/Assets/Scripts/Actor/ActorController.cs]: Spelling: optionally -> optional

File: unity-ggjj/Assets/Scripts/Actor/ActorController.cs [hunk 7]
```diff
@@ -108,53 +108,48 @@ private ActorData FindActorDataInInventory(string actorName)
...
+
+        _activeActor.PlayAnimation(pose);
     }
 
     /// <summary>
     /// Plays the animation of an actor by playing the specified animation on the actor's animator.
     /// Flags the system as busy so it waits for the animation to end.
     /// </summary>
     /// <param name="emotion">The emotion to play.</param>
+    /// <param name="actorName">Optionally name of another actor to run this animation on (defaults to <see cref="_activeActor"/> if not set)</param>
```
Comment (PhoebeMitchell) [inline unity-ggjj/Assets/Scripts/Actor/ActorController.cs]: Spelling: optionally -> optional

File: unity-ggjj/Assets/Scripts/Actor/ActorController.cs [hunk 8]
```diff
@@ -299,4 +288,16 @@ public void AssignActorToSlot(string actor, int oneBasedSlotIndex)
             Debug.Log($"{exception.GetType().Name}: Actor {actor} was not found in actor dictionary");
         }
     }
+
+    /// <summary>
+    /// Change the visibility of the sprite of the specified actor
+    /// </summary>
+    /// <param name="actorName">Name of the actor to change the visibility of</param>
+    /// <param name="shouldShow">Whether to show (`true`) or hide `false` the actor</param>
```
Comment (PhoebeMitchell) [inline unity-ggjj/Assets/Scripts/Actor/ActorController.cs]: Nit: No parenthesis around 'false', like with ('true')

File: unity-ggjj/Assets/Scripts/TextDecoder/ActionDecoder.cs [hunk 9]
```diff
@@ -508,21 +508,15 @@ private void ACTOR(ActorAssetName actorName)
         OnActionDone?.Invoke();
     }
 
-    /// <summary>Shows or hides the actor on the screen. Has to be re-done after switching a scene.</summary>
-    /// <param name="shouldShow">whether to show (`true`) or not show (`false`) an actor</param>
-    /// <example>&amp;SHOW_ACTOR:true</example>
-    /// <example>&amp;SHOW_ACTOR:false</example>
+    /// <summary>Shows or hides all actors in the scene. Has to be re-done after switching a scene.</summary>
+    /// <param name="actorName" validFiles="Assets/ScriptableObjects/Actors/*.asset">Name of the actor</param>
+    /// <param name="shouldShow">Whether to show (`true`) or not show (`false`) all actors</param>
```
Comment (PhoebeMitchell) [inline unity-ggjj/Assets/Scripts/TextDecoder/ActionDecoder.cs]: Would changing "all actors" to "current actor" be more accurate?

**Review by PhoebeMitchell — DISMISSED** [review]
All functional :D Couple of nits. Nothing block-worthy.

**Review by notexplosive — DISMISSED** [review]

**Review by lyndsiWilliams — APPROVED** [review]
Code looks good, tests pass, feature works as described!

**Review by PhoebeMitchell — APPROVED** [review]

**Pull request merged**